File: js/nova/datastruct/list/DoubleArray.js
var DoubleArray = function () {
	
	
};

DoubleArray.prototype.DoubleArray = function () {
	this.this();
};

DoubleArray.prototype.DoubleArray = function (count) {
	this.this(count);
};

DoubleArray.prototype.DoubleArray = function (data, count) {
	this.this(data, count);
};

DoubleArray.prototype.destroy = function () {
};

DoubleArray.prototype.this = function () {
	this.this();
};

DoubleArray.prototype.this = function (count) {
	this.this(count);
};

DoubleArray.prototype.this = function (data, count) {
	this.this(data, count);
};

DoubleArray.prototype.map = function (mapFunc) {
	var array;
	var i;
	var nova_local_0;
	var element;
	array = new Array();
	i = 0;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		array.add(mapFunc(element, i++, this));
	}
	return array;
};

DoubleArray.prototype.forEach = function (func) {
	var i;
	i = 0;
	for (; i < this.count; i++) {
		func(this.get(i), i, this);
	}
};

DoubleArray.prototype.any = function (anyFunc) {
	var nova_local_0;
	var element;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		if (anyFunc(element)) {
			return true;
		}
	}
	return false;
};

DoubleArray.prototype.all = function (allFunc) {
	var nova_local_0;
	var element;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		if (!allFunc(element)) {
			return false;
		}
	}
	return true;
};

DoubleArray.prototype.filter = function (filterFunc) {
	var filtered;
	var i;
	var nova_local_0;
	var element;
	filtered = new DoubleArray();
	i = 0;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		if (filterFunc(element, i++, this)) {
			filtered.add(element);
		}
	}
	return filtered;
};

DoubleArray.prototype.take = function (howMany) {
	var list;
	var i;
	if (howMany > this.count) {
		howMany = this.count;
	}
	list = new DoubleArray();
	i = 0;
	for (; i < howMany; i++) {
		list.add(this.data);
	}
	return list;
};

DoubleArray.prototype.skip = function (howMany) {
	var list;
	var i;
	list = new DoubleArray();
	i = howMany;
	for (; i < this.count; i++) {
		list.add(this.data);
	}
	return list;
};

DoubleArray.prototype.firstWhere = function (func) {
	var nova_local_0;
	var element;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		if (func(element)) {
			return element;
		}
	}
	return 0;
};

DoubleArray.prototype.reverse = function () {
	var array;
	var i;
	var nova_local_0;
	var element;
	array = new DoubleArray(this.count);
	array.count = this.count;
	i = 0;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		array.set(this.count - ++i, element);
	}
	return array;
};

DoubleArray.prototype.join = function (delimiter) {
	var str;
	var passed;
	var nova_local_0;
	var element;
	str = "";
	passed = false;
	nova_local_0 = (this).iterator();
	while (nova_local_0.hasNext()) {
		element = nova_local_0.next();
		if (passed) {
			str = str.concat(delimiter);
		} else {
			passed = true;
		}
		str = str.concat(Double.toString(element));
	}
	return str;
};

DoubleArray.prototype.accessor_iterator = function () {
	return new DoubleArrayIterator(this);
};

DoubleArray.prototype.mutator_iterator = function () {
};

DoubleArray.prototype.accessor_first = function () {
	if (this.count > 0) {
		return this.data;
	}
	return 0;
};

DoubleArray.prototype.mutator_first = function () {
};

DoubleArray.prototype.accessor_last = function () {
	if (this.count > 0) {
		return this.data;
	}
	return 0;
};

DoubleArray.prototype.super = function () {
};

File: java/nova/datastruct/list/DoubleArray.java
package nova.datastruct.list;

import java.util.Optional;
import nova.exception.ExceptionData;
import nova.exception.Exception;
import nova.exception.DivideByZeroException;
import nova.io.NovaConsole;
import nova.primitive.number.Number;
import nova.primitive.number.NovaByte;
import nova.primitive.number.NovaShort;
import nova.primitive.number.NovaInt;
import nova.primitive.number.NovaLong;
import nova.primitive.number.NovaFloat;
import nova.primitive.number.NovaDouble;
import nova.primitive.Null;
import nova.primitive.number.Char;
import nova.primitive.Bool;
import nova.datastruct.list.NovaArray;
import nova.datastruct.list.IntArray;
import nova.datastruct.list.CharArray;
import nova.datastruct.list.DoubleArray;
import nova.datastruct.list.IntRange;
import nova.thread.Thread;
import nova.thread.async.Async;
import nova.gc.GC;
import nova.math.NovaMath;
import nova.NovaObject;
import nova.NovaString;
import nova.System;
import nova.Class;
import nova.datastruct.list.DoubleArrayIterator;

public class DoubleArray extends NovaArray
{
	
	
	
	
	
	public DoubleArray()
	{
		init();
	}
	
	public DoubleArray(double[] data, int count)
	{
		init(data, count);
	}
	
	public DoubleArray(int count)
	{
		init(count);
	}
	
	public DoubleArrayIterator accessor_iterator()
	{
		return new DoubleArrayIterator(this);
	}
	
	private DoubleArrayIterator mutator_iterator()
	{
	}
	
	public double accessor_first()
	{
		return count > 0 ? this.get(0) : 0;
	}
	
	private double mutator_first()
	{
	}
	
	public double accessor_last()
	{
		return count > 0 ? this.get(count - 1) : 0;
	}
	
	private double mutator_last()
	{
	}
	
	public void init()
	{
		init();
	}
	
	public void init(double[] data, int count)
	{
		init(data, count);
	}
	
	public void init(int count)
	{
		init(count);
		
	}
	
	public double sum(NovaUtilities.Function3<Double, Int, DoubleArray, Double> func)
	{
		double sum;
		int i;
		DoubleArrayIterator nova_local_0;
		double element;
		sum = 0;
		i = 0;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			sum += func.call(element, i++, this);
		}
		return sum;
	}
	
	public NovaArray map(NovaUtilities.Function3<Double, Int, DoubleArray, Out> mapFunc)
	{
		NovaArray array;
		int i;
		DoubleArrayIterator nova_local_0;
		double element;
		array = new NovaArray();
		i = 0;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			array.add(mapFunc.call(element, i++, this));
		}
		return array;
	}
	
	public DoubleArray forEach(NovaUtilities.Consumer3<Double, Int, DoubleArray> func)
	{
		int i;
		i = (int)0;
		for (; i < (int)count; i++)
		{
			func.call((double)get(i), i, this);
		}
		return this;
	}
	
	public boolean any(NovaUtilities.Function1<Double, Bool> anyFunc)
	{
		DoubleArrayIterator nova_local_0;
		double element;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			if (anyFunc.call(element))
			{
				return true;
			}
		}
		return false;
	}
	
	public boolean all(NovaUtilities.Function1<Double, Bool> allFunc)
	{
		DoubleArrayIterator nova_local_0;
		double element;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			if (!allFunc.call(element))
			{
				return false;
			}
		}
		return true;
	}
	
	public DoubleArray filter(NovaUtilities.Function3<Double, Int, DoubleArray, Bool> filterFunc)
	{
		DoubleArray filtered;
		int i;
		DoubleArrayIterator nova_local_0;
		double element;
		filtered = new DoubleArray();
		i = 0;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			if (filterFunc.call(element, i++, this))
			{
				filtered.add(element);
			}
		}
		return filtered;
	}
	
	public DoubleArray take(int howMany)
	{
		DoubleArray list;
		int i;
		if (howMany > count)
		{
			howMany = count;
		}
		list = new DoubleArray();
		i = (int)0;
		for (; i < (int)howMany; i++)
		{
			list.add(get(i));
		}
		return list;
	}
	
	public DoubleArray skip(int howMany)
	{
		DoubleArray list;
		int i;
		list = new DoubleArray();
		i = (int)howMany;
		for (; i < (int)count; i++)
		{
			list.add(get(i));
		}
		return list;
	}
	
	public double firstWhere(NovaUtilities.Function1<Double, Bool> func)
	{
		DoubleArrayIterator nova_local_0;
		double element;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			if (func.call(element))
			{
				return element;
			}
		}
		return 0;
	}
	
	public DoubleArray reverse()
	{
		DoubleArray array;
		int i;
		DoubleArrayIterator nova_local_0;
		double element;
		array = new DoubleArray(count);
		array.count = count;
		i = 0;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			array.set(count - ++i, element);
		}
		return array;
	}
	
	public NovaString join(NovaString delimiter)
	{
		NovaString str;
		boolean passed;
		DoubleArrayIterator nova_local_0;
		double element;
		str = new NovaString("");
		passed = false;
		nova_local_0 = (this).accessor_iterator();
		while (nova_local_0.accessor_hasNext())
		{
			element = nova_local_0.accessor_next();
			if (passed)
			{
				str = str.concat(delimiter);
			}
			else
			{
				passed = true;
			}
			str = str.concat(NovaDouble.toString(element));
		}
		return str;
	}
	
	public double get(int index)
	{
		return ((double[])data)[index];
	}
	
	public double set(int index, double value)
	{
		((double[])data)[index] = value;
		return value;
	}
	
}
